package.json

```diff
@@ -20,11 +20,12 @@
     "start": "npm run serve-examples",
     "serve-examples": "webpack serve --config examples/webpack/config.js --mode development",
     "build-examples": "webpack --config examples/webpack/config.js --mode production",
-    "build-package": "npm run transpile && npm run copy-css && node tasks/prepare-package.js",
+    "build-package": "npm run transpile && npm run copy-css && npm run move-types && node tasks/prepare-package.js",
     "build-index": "shx rm -f build/index.js && npm run build-package && node tasks/generate-index.js",
     "build-legacy": "shx rm -rf build/legacy && npm run build-index && webpack --config config/webpack-config-legacy-build.js && cleancss --source-map src/ol/ol.css -o build/legacy/ol.css",
     "build-site": "npm run build-examples && npm run apidoc && mkdir -p build/site && cp site/index.html build/site && mv build/apidoc build/site/apidoc && mv build/examples build/site/examples",
     "copy-css": "shx cp src/ol/ol.css build/ol/ol.css",
+    "move-types": "shx rm -rf build/types && shx mkdir -p build/types && shx cp -Rf build/ol/* build/types/ && shx rm 'build/ol/**/*.d.ts*' && shx rm 'build/types/**/*.js*' && shx mv build/types build/ol/",
```

Comment:
maybe use `--emitDeclaration=false` and `--emitDeclaration=true` with `--outDir`?

Reply:
sorry wrong syntax. but there was a way to do this via cli.

Reply:
Maybe. I also need to touch the declaration maps so they point to the correct source files. The way it is now, users will be taken to the type definition instead of the source when they click "Go to definition" in VS Code.

Reply:
with these options it should be possible to create to different commands, one for typechecking the other for creating the declaration. Maybe `declaration` needs to be set to false in the `tscconfig` then:

`tsc --declaration --emitDeclarationOnly --noEmit false --outDir xyz`

Reply:
ah well. Just turn off `declaration` and `declarationMap` again and use:
`tsc --declaration --declarationMap --outDir xyz`